src/browser/page-switch.js

```diff
@@ -32,7 +29,13 @@ define((require, exports, moudle) => {
           remove(items, equals(item));
         }
       }
-    }));
+    }, [
+      DOM.span({
+        key: `thumb-${item.get('id')}`,
```

Comment:
Nit: You don't need to include id in the key, keys just need to be unique with-in it's siblings not across the whole tree.